src/screens/TokensListScreen.tsx

```diff
@@ -0,0 +1,179 @@
+import { StyleSheet, View, ScrollView } from 'react-native'
+import React, { Fragment, useCallback, useMemo, useState } from 'react'
+import { NativeStackScreenProps } from 'react-native-screens/native-stack'
+import { TextInput, Text, Divider, FAB, Portal, List } from 'react-native-paper'
+import { useIsFocused } from '@react-navigation/core'
+
+import { MainStackParamList } from '../Main'
+import theme from '../lib/theme'
+import { useSecretSelector } from '../hooks/use-secret-selector'
+import { Typography } from '../components/Typography'
+
+const styles = StyleSheet.create({
+  container: {
+    flexGrow: 1,
+    justifyContent: 'flex-start',
+    paddingHorizontal: theme.spacing(2),
+    paddingVertical: theme.spacing(3),
+  },
+  noTokens: {
+    marginTop: theme.spacing(8),
+    padding: theme.spacing(4),
+  },
+  noTokensTitle: {
+    textAlign: 'center',
+    marginBottom: theme.spacing(1),
+  },
+  noTokensBody: {
+    textAlign: 'center',
+  },
+  searchArea: {
+    marginBottom: theme.spacing(2),
+  },
+  scrollView: {
+    flexGrow: 1,
+  },
+  tokensCount: {
+    marginBottom: theme.spacing(2),
+  },
+  tokensCountLabel: {
+    fontFamily: 'monospace',
+    fontSize: 24,
+    marginRight: theme.spacing(2),
+  },
+  tokensCountValue: theme.typography.overline,
+  tokenItem: {
+    paddingHorizontal: 0,
+    paddingVertical: theme.spacing(1),
+    // React native paper doesn't allow overriding this https://github.com/callstack/react-native-paper/blob/b545cdcbd8c5f1bd5ad3f0e9f095c294527846a4/src/components/List/ListItem.tsx#L263
+    // Needed to override the padding that's hardcoded there
+    marginLeft: -8,
+  },
+  tokenValueText: {
+    fontFamily: 'monospace',
+    fontSize: 24,
+    color: theme.colors.text,
+    marginBottom: theme.spacing(1),
+  },
+  tokenDescriptionText: {
+    ...theme.typography.body2,
+    color: theme.colors.text,
+    marginHorizontal: 0,
+  },
+  fab: {
+    backgroundColor: theme.colors.primary,
+    position: 'absolute',
+    margin: theme.spacing(3),
+    bottom: 0,
+    alignSelf: 'center',
+  },
+})
+
+type Props = NativeStackScreenProps<MainStackParamList, 'TokensList'>
+
+export const TokensListScreen = ({ route, navigation }: Props) => {
+  const { navigate } = navigation
+  const { secretId } = route.params
+  const secret = useSecretSelector(secretId)
+  const tokens = useMemo(() => (secret.tokens ? secret.tokens : []), [secret])
+  const tokensCount = tokens.length
+  const [search, setSearch] = useState('')
+  const [searchFocused, setSearchFocused] = useState(false)
+  const isFocused = useIsFocused()
+  const shouldShowFab = isFocused && !searchFocused
+
+  const filteredTokens = useMemo(() => {
+    if (search.length > 1) {
+      return tokens.filter(({ description }) =>
```

Comment:
If the filter result is zero, could we filter by `token`?

For example: I see a token on my CI, and I want to check that the description match because I don't remember if I have copied&pasted it from another setup

Reply:
I've created a separate issue for this. See here: https://github.com/nearform/optic-expo/issues/262